app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt

```diff
@@ -23,16 +25,16 @@ class ToolbarIntegration(
     toolbar: BrowserToolbar,
     domainAutocompleteProvider: DomainAutocompleteProvider,
     historyStorage: HistoryStorage,
+    sessionManager: SessionManager,
     sessionId: String? = null
 ) : LifecycleObserver {
     init {
         toolbar.setMenuBuilder(context.components.toolbar.menuBuilder)
 
         val home = BrowserToolbar.Button(
-            context.resources.getDrawable(
-                R.drawable.ic_home,
-                context.application.theme
-            ), context.getString(R.string.browser_home_button)
+            context.resources.getDrawable(R.drawable.ic_home, context.application.theme),
+            context.getString(R.string.browser_home_button),
+            visible = { sessionManager.runWithSession(sessionId) { it.isCustomTabSession().not() } }
```

Comment:
Instead of passing in the entire sessionManager, lets pass in a function that will determine whether or not to show the home button.

Reply:
Do you mean expose this visibility lambda in the `ToolbarIntegration` constructor?

Reply:
Yep :+1:

Reply:
Sure, I can do that, although it seems odd to me to expose that outside of the integration class for this specific button. Otherwise, we would have to pass the `sessionId` to other use cases (`ToolbarUIView`).

I can see that the `ToolbarFeature` below is accessing the `SessionManager` from the context below. I can use the `SessionManager` in a similar way, would that work?

Reply:
That is a very good point. I think I'd rather keep your solution in this case. I'm not sure how much I like everything needing access to `requireComponents` :)

Reply:
> Otherwise, we would have to pass the sessionId to other use cases (ToolbarUIView).

Actually, this is incorrect. Even in the new `ToolbarUIView` we _need_ the sessionId passed to it, if this would be the new implementation for it, otherwise the toolbar would never know where to get the `CustomTabConfig` in it's session.

I can file an issue as a follow up.